backend/restapi/serializer.py

```diff
@@ -1,16 +1,38 @@
 from rest_framework import serializers
 from rest_framework.exceptions import NotFound
+from django.db.models import Sum
 from .models import Allowed_topic_names, Mission, Topic
 from .models import File
 from .models import Tag
 from .models import Mission_tags
 
 
 class MissionSerializer(serializers.ModelSerializer):
+    total_duration = serializers.SerializerMethodField()
+    total_size = serializers.SerializerMethodField()
+    robots = serializers.SerializerMethodField()
+
     class Meta:  # definition of which data to serialize
         model = Mission
         fields = "__all__"
 
+    def get_total_duration(self, obj):
+        # calculate the total duration of all files in the mission
+        result = File.objects.filter(mission=obj).aggregate(Sum("duration"))
+        return result["duration__sum"] or 0
+
+    def get_total_size(self, obj):
+        # calculate the total size of all files in the mission
+        result = File.objects.filter(mission=obj).aggregate(Sum("size"))
+        return result["size__sum"] or 0
+
+    def get_robots(self, obj):
+        # get all robot names in the mission
+        result = (
+            File.objects.filter(mission=obj).values_list("robot", flat=True).distinct()
```

Comment:
```suggestion
            list(File.objects.filter(mission=obj).values_list("robot", flat=True).distinct())
```
If you wrap this in list() you don't have those problems where it becomes `<QuerySet[...]>` which is also visible when you run `python cli.py mission list` and it always is a list.

Reply:
You could also already make it a comma separated list as a string with `join` and maybe add a check for `None` to remove `None` from the list

Reply:
Good Idea. I will do that